### ISSUE TITLE
Issue with TSS enrichment

### ISSUE THREAD

**Issue by menage95** [stuart-lab/signac#1546]
Hello, 
TSSEnrichment() function is suddenly throwing up an error despite the same code being run before without issue. The only difference was that I had to reinstall SeuratObject().

```
data.peaks <- TSSEnrichment(data.peaks, fast = FALSE)

Extracting TSS positions
Finding + strand cut sites
Finding - strand cut sites
Computing mean insertion frequency in flanking regions
Normalizing TSS score
Error in SetAssayData.ChromatinAssay(object = object[[assay]], layer = layer,  : 
  argument "slot" is missing, with no default

sessionInfo()
R version 4.3.1 (2023-06-16 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22621)

Matrix products: default


locale:
[1] LC_COLLATE=English_United Kingdom.utf8  LC_CTYPE=English_United Kingdom.utf8    LC_MONETARY=English_United Kingdom.utf8
[4] LC_NUMERIC=C                            LC_TIME=English_United Kingdom.utf8    

time zone: Europe/London
tzcode source: internal

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] BSgenome.Mmusculus.UCSC.mm10_1.4.3 BSgenome_1.68.0                    rtracklayer_1.60.1                
 [4] Biostrings_2.68.1                  XVector_0.40.0                     dplyr_1.1.3                       
 [7] qs_0.25.5                          biovizBase_1.48.0                  hdf5r_1.3.8                       
[10] patchwork_1.1.3                    EnsDb.Mmusculus.v79_2.99.0         ensembldb_2.24.1                  
[13] AnnotationFilter_1.24.0            GenomicFeatures_1.52.2             AnnotationDbi_1.62.2              
[16] Biobase_2.60.0                     GenomicRanges_1.52.0               GenomeInfoDb_1.36.4               
[19] IRanges_2.34.1                     S4Vectors_0.38.1                   BiocGenerics_0.46.0               
[22] ggplot2_3.4.4                      readr_2.1.4                        scCustomize_2.0.1                 
[25] SeuratObject_5.0.1                 Seurat_4.3.0.1                     Signac_1.10.0                     

loaded via a namespace (and not attached):
  [1] ProtGenerics_1.32.0         matrixStats_1.0.0           spatstat.sparse_3.0-2       bitops_1.0-7               
  [5] lubridate_1.9.3             httr_1.4.7                  RColorBrewer_1.1-3          tools_4.3.1                
  [9] sctransform_0.3.5           backports_1.4.1             utf8_1.2.3                  R6_2.5.1                   
 [13] lazyeval_0.2.2              uwot_0.1.16                 withr_2.5.2                 sp_2.0-0                   
 [17] prettyunits_1.2.0           gridExtra_2.3               progressr_0.14.0            cli_3.6.1                  
 [21] spatstat.explore_3.2-3      spatstat.data_3.0-3         ggridges_0.5.4              pbapply_1.7-2              
 [25] Rsamtools_2.16.0            foreign_0.8-84              dichromat_2.0-0.1           parallelly_1.36.0          
 [29] rstudioapi_0.15.0           RSQLite_2.3.1               RApiSerialize_0.1.2         generics_0.1.3             
 [33] shape_1.4.6                 BiocIO_1.10.0               ica_1.0-3                   spatstat.random_3.1-5      
 [37] Matrix_1.6-3                ggbeeswarm_0.7.2            fansi_1.0.4                 abind_1.4-5                
 [41] lifecycle_1.0.4             yaml_2.3.7                  snakecase_0.11.1            SummarizedExperiment_1.30.2
 [45] BiocFileCache_2.8.0         Rtsne_0.16                  paletteer_1.5.0             grid_4.3.1                 
 [49] blob_1.2.4                  promises_1.2.1              crayon_1.5.2                miniUI_0.1.1.1             
 [53] lattice_0.21-8              cowplot_1.1.1               KEGGREST_1.40.1             pillar_1.9.0               
 [57] knitr_1.45                  rjson_0.2.21                future.apply_1.11.0         codetools_0.2-19           
 [61] fastmatch_1.1-4             leiden_0.4.3.1              glue_1.6.2                  data.table_1.14.8          
 [65] vctrs_0.6.3                 png_0.1-8                   spam_2.10-0                 gtable_0.3.4               
 [69] rematch2_2.1.2              cachem_1.0.8                xfun_0.40                   S4Arrays_1.0.6             
 [73] mime_0.12                   survival_3.5-5              RcppRoll_0.3.0              ellipsis_0.3.2             
 [77] fitdistrplus_1.1-11         ROCR_1.0-11                 nlme_3.1-162                bit64_4.0.5                
 [81] progress_1.2.2              filelock_1.0.2              RcppAnnoy_0.0.21            irlba_2.3.5.1              
 [85] vipor_0.4.5                 KernSmooth_2.23-21          rpart_4.1.19                colorspace_2.1-0           
 [89] DBI_1.1.3                   Hmisc_5.1-1                 nnet_7.3-19                 ggrastr_1.0.2              
 [93] tidyselect_1.2.0            bit_4.0.5                   compiler_4.3.1              curl_5.0.2                 
 [97] htmlTable_2.4.2             xml2_1.3.5                  DelayedArray_0.26.7         plotly_4.10.3              
[101] stringfish_0.15.8           checkmate_2.2.0             scales_1.2.1                lmtest_0.9-40              
[105] rappdirs_0.3.3              stringr_1.5.1               digest_0.6.33               goftest_1.2-3              
[109] spatstat.utils_3.0-3        rmarkdown_2.25              htmltools_0.5.6             pkgconfig_2.0.3            
[113] base64enc_0.1-3             MatrixGenerics_1.12.3       dbplyr_2.4.0                fastmap_1.1.1              
[117] rlang_1.1.1                 GlobalOptions_0.1.2         htmlwidgets_1.6.2           shiny_1.8.0                
[121] zoo_1.8-12                  jsonlite_1.8.7              BiocParallel_1.34.2         VariantAnnotation_1.46.0   
[125] RCurl_1.98-1.12             magrittr_2.0.3              Formula_1.2-5               GenomeInfoDbData_1.2.10    
[129] dotCall64_1.1-0             munsell_0.5.0               Rcpp_1.0.11                 reticulate_1.31            
[133] stringi_1.7.12              zlibbioc_1.46.0             MASS_7.3-60                 plyr_1.8.8                 
[137] parallel_4.3.1              listenv_0.9.0               ggrepel_0.9.3               forcats_1.0.0              
[141] deldir_1.0-9                splines_4.3.1               tensor_1.5                  hms_1.1.3                  
[145] circlize_0.4.15             igraph_1.5.1                spatstat.geom_3.2-5         reshape2_1.4.4             
[149] biomaRt_2.56.1              XML_3.99-0.14               evaluate_0.23               RcppParallel_5.1.7         
[153] ggprism_1.0.4               tzdb_0.4.0                  httpuv_1.6.11               RANN_2.6.1                 
[157] tidyr_1.3.0                 purrr_1.0.2                 polyclip_1.10-4             future_1.33.0              
[161] scattermore_1.2             janitor_2.2.0               xtable_1.8-4                restfulr_0.0.15            
[165] later_1.3.1                 viridisLite_0.4.2           tibble_3.2.1                memoise_2.0.1              
[169] beeswarm_0.4.0              GenomicAlignments_1.36.0    cluster_2.1.4               timechange_0.2.0           
[173] globals_0.16.2  

```

Thanks for any help!

**Comment by timoast** [comment]
Since you're using SeuratObject v5 you need to update Seurat and Signac to the latest release

**Comment by menage95** [comment]
The PercentageSetFeatures() is also no longer working:

```
Backtrace:
    ▆
 1. └─Seurat::PercentageFeatureSet(het.peaks, pattern = "^mt-")
 2.   ├─features %||% ...
 3.   ├─base::grep(...)
 4.   │ └─base::is.factor(x)
 5.   ├─base::rownames(x = object[[assay]][layer])
 6.   ├─object[[assay]][layer]
 7.   └─SeuratObject:::`[.Assay`(object[[assay]], layer)
 8.     ├─SeuratObject::LayerData(...)
 9.     └─SeuratObject:::LayerData.Assay(features = i)
Run rlang::last_trace(drop = FALSE) to see 9 hidden frames.
```

**Comment by menage95** [comment]
@millerh1 

I managed to fix this by installing the development version of signac - try and see if it works for you

**Comment by edfajardo** [comment]
> @millerh1
> 
> I managed to fix this by installing the development version of signac - try and see if it works for you

Thanks so much for this.  It worked!

**Comment by cnluzon** [comment]
I experienced the same issue and fixed by installing the development version as @menage95 suggested (thanks!), so I guess it will be solved once the next version is released. However I also agree this should be open at least so others can find this workaround easier!

```
remotes::install_github("stuart-lab/signac", ref = "develop")
```

I ran into this issue by going through pbmc vignette: https://stuartlab.org/signac/articles/pbmc_vignette, so it is something that happens with sample data.
